test/unit/spa.test.js

```diff
@@ -82,6 +82,23 @@ describe('spa', () => {
     expect(consola.log).toHaveBeenCalledWith('mounted')
     consola.log.mockClear()
   })
+
+  test('/async no asyncData leak', async () => {
+    const window = await nuxt.server.renderAndGetWindow(url('/async'))
+
+    const navigate = url => new Promise((resolve, reject) => {
+      window.$nuxt.$router.push(url, resolve, reject)
+    })
+
+    for (let i = 0; i < 3; i++) {
+      await navigate('/')
+      await navigate('/async')
+    }
+
+    const { $data } = window.$nuxt.$route.matched[0].instances.default
+    expect(Object.keys($data).length).toBe(1)
```

Comment:
The data of the old page would be better to destroyed by vue itself (and it should) the problem is that we _chain_ async data over and over again.

Reply:
I know about chain problem, but in this PR setting `Component.options.__isNuxt = true` and in next time calling applyAsyncData - return if __isNuxt = true - will not set new asyncData 😞and therefore this will solve chain problem, but not solve setting correct asyncData (I just guessing and not checked this)

Reply:
@yariksav i guess you are experimenting on `nuxt dev` which has enabled `runInNewContext` so your experiments should go wrong. You can test with prob start or disabling via options.

Reply:
```js
{
  render: {
    runInNewContext: false
  }
}
```